src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const actionsArray = [];
```

Comment:
The name `actionsArray` can be improved to reflect its purpose more clearly. Consider renaming it to `stateHistory` or `states`, as it represents the history of states after each action.